### PR TITLE
Refresh readme

### PR DESCRIPTION
**Pull request by willingc**
Refreshing 🍹

### CODE REVIEW

**Comment by willingc** [comment]
A few questions:
- looking at the versions of releases for badges, it looks like npm has a 1.x version. do you want to deploy a more recent?
- Can the installation for devs be simplified by using zmq-prebuilt?

File: README.md [hunk 1]
```diff
@@ -1,91 +1,101 @@
 # enchannel-zmq-backend
 
-![enchannel version](https://img.shields.io/badge/enchannel-1.1-ff69b4.svg)
+[![GitHub release](https://img.shields.io/badge/version-latest-blue.svg)](https://github.com/nteract/enchannel-zmq-backend/releases)
```
Comment (rgbkrk) [inline README.md]: This "enchannel version" in reference to the "spec" in [enchannel](https://github.com/nteract/enchannel), which isn't so much a spec as "please make `Rx.Subject`s" for each of the Jupyter channels. That being said, I think we should switch to this format you have here.

File: README.md [hunk 2]
```diff
@@ -1,91 +1,101 @@
...
-#### OS X
+## Technical overview
 
-##### homebrew on OS X
+As a refresher for the reader, *enchannel* details nteract's
+lightweight, implementation-flexible specification for communication
+between a user frontend and a backend, such as a language kernel. The 
+*enchannel* specification offers a simple description of "what" 
+messages may be passed between frontends and backends, while leaving
+a developer freedom in "how" to achieve message communication.
```
Comment (rgbkrk) [inline README.md:20]: 👍

File: README.md [hunk 3]
```diff
@@ -1,91 +1,101 @@
...
 
-- [`pkg-config`](http://www.freedesktop.org/wiki/Software/pkg-config/): `brew install pkg-config`
-- [ZeroMQ](http://zeromq.org/intro:get-the-software): `brew install zeromq`
+**enchannel-zmq-backend** takes a classic design approach using ZeroMQ,
+the foundation messaging protocol for the Jupyter project.
```
Comment (rgbkrk) [inline README.md:23]: foundation - primo word choice

File: README.md [hunk 4]
```diff
@@ -1,91 +1,101 @@
...
+enchannel-zmq-backend implements backend support for the messaging
+channels described in the [Jupyter messaging specification][]. This
```
Comment (rgbkrk) [inline README.md:25]: Probably want to link to http://jupyter-client.readthedocs.io/en/latest/messaging.html here.

Comment (rgbkrk) [reply README.md:25]: Oh wait, that's what you're doing down below. Silly me.

File: README.md [hunk 5]
```diff
@@ -1,91 +1,101 @@
...
-#### Linux
-
-For Debian/Ubuntu based variants, you'll need `libzmq3-dev` (preferred) or alternatively `libzmq-dev`.   
-For RedHat/CentOS/Fedora based variants, you'll need `zeromq` and `zeromq-devel`.
-
-## About
-
-Provides functions to create [RxJS](https://github.com/ReactiveX/RxJS) subjects (`Observable`s and `Subscriber`s) for four of the Jupyter channels:
+We provide functions to create [RxJS](https://github.com/ReactiveX/RxJS) 
+subjects (`Observable` and `Subscriber` subjects) for four of the
```
Comment (rgbkrk) [inline README.md]: ...subjects (two-way Observables)...

File: README.md [hunk 6]
```diff
@@ -1,91 +1,101 @@
...
+      ip: '127.0.0.1',
+      control_port: 58787,
+      hb_port: 58788,
+      signature_scheme: 'hmac-sha256',
+      key: 'dddddddd-eeee-aaaa-dddd-dddddddddddd',
+      shell_port: 58784,
+      transport: 'tcp',
+      iopub_port: 58785
+    }
+    ```
```
Comment (rgbkrk) [inline README.md:92]: spawnteract can be listed here as a helper for launching the kernel.

File: README.md [hunk 7]
```diff
@@ -146,7 +157,8 @@ var message = {
 };
 ```
 
-Until we make changes, you'll need to have at least one subscription before you can send on a channel.
+Currently, you'll need to have at least one subscription activated
+before you can send on a channel.
```
Comment (rgbkrk) [inline README.md:166]: Oooh! I have a pattern for this now as part of `nteract/nteract`.

They _can_ subscribe then send. However, they can also wrap sending in as part of their Observable:

``` js
// The message being sent
const message = {...}
// The observable you'd subscribe to later
const original = iopub.filter(...);

// On subscription, send the message
return Rx.Observable.create(observer => {
  const subscription = original.subscribe(observer);
  channels.shell.next(message);
  return subscription;
});
```

We do need to make our own Subject type, as suggested by @jayphelps, but I haven't been able to come back to it yet.

File: README.md [hunk 8]
```diff
@@ -179,3 +191,55 @@ Until we make changes, you'll need to have at least one subscription before you
...
      payload: [] } }
 ```
+
+## Contributors and developers
+
+### ZeroMQ Dependency
+
+If you plan to contribute to this project or extend it, you will need
+to have ZeroMQ installed on your system.
+
```
Comment (rgbkrk) [inline README.md:208]: Oh! We rely on `zmq-prebuilt` here now so technically they don't have to go through all this yucky setup anymore.

**Review by rgbkrk — APPROVED** [review]

**Comment by rgbkrk** [comment]
> Can the installation for devs be simplified by using zmq-prebuilt?

Yes!

**Comment by rgbkrk** [comment]
> looking at the versions of releases for badges, it looks like npm has a 1.x version. do you want to deploy a more recent?

Is this about the enchannel badge or another one?

File: README.md (hunk 8 above)
Comment (willingc) [reply README.md:208]: Can this just be deleted fully?

Comment (rgbkrk) [reply README.md:208]: Yes.

**Comment by rgbkrk** [comment]
Thank you so much for this refresher!

**Comment by willingc** [comment]
If I go to npm site and search for enchannel-zmq-backend, I see:
screen shot 2016-09-24 at 2 18 30 pm

I see in the upper left enchannel as 1.1 and on the right hand side as 2.3.1. Not sure what the difference is???

**Comment by rgbkrk** [comment]
That `1.1` is our spec number, which looks very silly above. We can just delete that badge.

**Comment by willingc** [comment]
Got it the 1.1 comes from the `enchannel` repo. I may be able to make a custom badge that clarifies or change the layout of the readme slightly to capture both versions.

File: README.md (hunk 1 above)
Comment (willingc) [reply README.md]: Updated 2 badges - one for spec and one for repo enchained-zmq-backend

File: README.md (hunk 5 above)
Comment (willingc) [reply README.md]: Double check the updates and links that I made here. Thanks!

File: README.md (hunk 7 above)
Comment (willingc) [reply README.md:166]: Let's leave as is now and then update when work is finished.

File: README.md (hunk 6 above)
Comment (willingc) [reply README.md:92]: Oops need to loop back on this one. Missed it on first pass.

**Comment by willingc** [comment]
Thanks for the merge. One more change coming in another PR 👍

**Comment by rgbkrk** [comment]
yay!

**Comment by willingc** [comment]
Rock on :guitar:

File: README.md (hunk 7 above)
Comment (rgbkrk) [reply README.md:166]: That sounds good. I look forward to making these more ergonomic in the future

**Pull request merged**